lambda/go.mod

```diff
@@ -1,11 +1,14 @@
 module cdk-ecr-deployment-handler
 
-go 1.15
+go 1.21
```

Comment:
revert or explain why this is needed

Reply:
Go mod tidy seemed to change this automatically. Build uses `public.ecr.aws/sam/build-go1.x:latest` which has go 1.22.1, but have to check if this works as should with Go 1.15.

Reply:
Seems that aws-sdk-go-v2 requires (according https://github.com/aws/aws-sdk-go-v2) Go 1.21. Not sure how those earlier ones have worked, or then this is always built with newer version than mentioned to be requirement on go.mod.

Reply:
Okay, you can probably guess my request to pull this update out into a separate PR 🙈

Sorry that this issue is bringing up so many maintenance tasks.

Reply:
OK, running those commands to check go.mod with older golang, as newer ones would update that automatically including also "toolchain" version there.